### PR TITLE
[ADD] fieldservice_purchase

### PR DESCRIPTION
**Pull request by osi-scampbell**
Odoo Purchases App and the Field Service App do not currently have a way to link FSM Workers
to their corresponding Pricelist.

This module allows you to link a Field Service Worker back to their corresponding Vendor Pricelist(s).

### CODE REVIEW

File: fieldservice_purchase/models/fsm_person.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    pricelist_count = fields.Integer(
+        compute='_compute_pricelist_count',
+        string='# Pricelists'
+    )
+
+    @api.multi
+    def _compute_pricelist_count(self):
+        for worker in self:
+            worker.pricelist_count = self.env['product.supplierinfo'].\
+                search_count([('name', '=', worker.name)])
```
Comment (max3903) [inline fieldservice_purchase/models/fsm_person.py]:
```suggestion
                search_count([('id', '=', worker.partner_id)])
```

File: fieldservice_purchase/models/fsm_person.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    @api.multi
+    def action_view_pricelists(self):
+        for worker in self:
+            pricelist = self.env['product.supplierinfo'].search(
+                [('name', '=', worker.name)])
```
Comment (max3903) [inline fieldservice_purchase/models/fsm_person.py]:
```suggestion
                [('id', '=', worker.partner_id)])
```

File: fieldservice_purchase/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    'license': 'AGPL-3',
+    'version': '12.0.1.0.0',
+    'depends': [
+        'fieldservice',
+        'purchase',
+    ],
+    'data': [
+        'views/fsm_person.xml',
+    ],
+    'installable': True,
```
Comment (max3903) [inline fieldservice_purchase/__manifest__.py]:
```suggestion
    'development_status': 'Beta',
```

For the installation, please add an option in the settings panel.

Comment (murtuzasaleh) [reply fieldservice_purchase/__manifest__.py]: @max3903 `For the installation, please add an option in the settings panel.` where?

Comment (max3903) [reply fieldservice_purchase/__manifest__.py]: In fieldservice:
https://github.com/OCA/field-service/blob/12.0/fieldservice/views/res_config_settings.xml

File: fieldservice_purchase/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (max3903) [inline fieldservice_purchase/models/__init__.py:2]:
```suggestion
# Copyright (C) 2019 - TODAY, Open Source Integrators
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```

**Review by max3903 — APPROVED** [review]

File: fieldservice_purchase/views/fsm_person.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (patrickrwilson) [inline fieldservice_purchase/views/fsm_person.xml]: This is no long er needed in v12 and the xml is missing copyright/license info.

**Review by patrickrwilson — APPROVED** [review]
Small change on xml.

**Comment by brian10048** [comment]
@osi-scampbell  
Can you also make this module enable the `group_manage_vendor_price` from Purchase settings in order to enable the pricelist menus in Purchases app by default?

On one of the runbot builds I created a new FSM Worker from Field Service app > Master Data > Workers menu.  Upon save, the pricelist count goes to 1.  It seems to set a random pricelist of a random vendor? 

Also, do you think this should the `supplier` field to True of the `res.partner` model associated to the FSM worker?

**Comment by patrickrwilson** [comment]
@brian10048 
> Also, do you think this should the `supplier` field to True of the `res.partner` model associated to the FSM worker?

I don't think this is a good idea because not every worker will be a supplier/vendor, some would be employees or ones that may not be classified as a supplier or vendor. However its probably a good idea to place the Is Vendor book field on the worker form to easily check that option if it's not already done so in the code.

**Comment by brian10048** [comment]
> 
> > Also, do you think this should the `supplier` field to True of the `res.partner` model associated to the FSM worker?
> 
> I don't think this is a good idea because not every worker will be a supplier/vendor, some would be employees or ones that may not be classified as a supplier or vendor. However its probably a good idea to place the Is Vendor book field on the worker form to easily check that option if it's not already done so in the code.

@patrickrwilson You are right, a field on the FSM worker would be a better option

**Pull request merged**